rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,140 @@
+
+    private HashSet<String> filesToDownload = new HashSet<String>();
+
+    @Autowired
+    private KustomizationFileReader kustomizationFileReader;
+
+    public BakeRecipe buildBakeRecipe(BakeManifestEnvironment env, KustomizeBakeManifestRequest request) {
+        BakeRecipe result = new BakeRecipe();
+        result.setName(request.getOutputName());
+        filesToDownload = new HashSet<>();
+        Artifact artifact = request.getInputArtifact();
+        if (artifact == null) {
+            throw new IllegalArgumentException("At least one input artifact must be provided to bake");
+        }
+        String kustomizationfilename = FilenameUtils.getName(artifact.getReference());
+        if (kustomizationfilename != null && !kustomizationfilename.toUpperCase().contains("KUSTOMIZATION")) {
+            throw new IllegalArgumentException("The inputArtifact should be a kustomization file valid.");
+        }
+
+        List<Artifact> artifacts = new ArrayList<>();
+        String templatePath = env.getStagingPath().resolve(FilenameUtils.getPath(artifact.getReference())).toString();
+        try {
+            Path path = Paths.get(FilenameUtils.getPath(artifact.getReference()));
+            artifact.setReference(path.toString());
+            HashSet<String> files = getFilesFromArtifact(artifact);
+            artifacts = files.stream()
+              .map(f -> {
+                return Artifact.builder()
+                    .reference(f)
+                    .artifactAccount(artifact.getArtifactAccount())
+                    .customKind(artifact.isCustomKind())
+                    .location(artifact.getLocation())
+                    .metadata(artifact.getMetadata())
+                    .name(artifact.getName())
+                    .provenance(artifact.getProvenance())
+                    .type(artifact.getType())
+                    .version(artifact.getVersion())
+                    .build();
+              }).collect(Collectors.toList());
+        } catch (IOException e) {
+            log.error("Error setting references in artifacts  " + e.getMessage());
+        }
+        try {
+            for (Artifact ar : artifacts) {
+                downloadArtifactToTmpFileStructure(env, ar);
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to fetch kustomize files: " + e.getMessage(), e);
+        }
+
+        List<String> command = new ArrayList<>();
+            command.add("kustomize");
+            command.add("build");
+            command.add(templatePath);
+        result.setCommand(command);
+
+        return result;
+    }
+
+    private HashSet<String> getFilesFromArtifact(Artifact artifact) throws IOException {
+        Path artifactPath = Paths.get(artifact.getReference());
+        Kustomization kustomization = kustomizationFileReader.getKustomization(artifact);
+        filesToDownload.add(kustomization.getReference());
```

Comment:
Use a new `HashSet<>` that you return at the end of the method.

Reply:
done